tests/src/test/scala/scalaz/MonadErrorTest.scala

```diff
@@ -8,38 +8,38 @@ import Isomorphism.{ <~>, IsoFunctorTemplate }
 object MonadErrorTest extends SpecLite {
 
   trait Decoder[A] {
-    def decode(s: String): String \/ A
+    def decode(s: String): Int \/ A
   }
   object Decoder {
     @inline def apply[A](implicit A: Decoder[A]): Decoder[A] = A
-    @inline def instance[A](f: String => String \/ A): Decoder[A] = new Decoder[A] {
-      override def decode(s: String): String \/ A = f(s)
+    @inline def instance[A](f: String => Int \/ A): Decoder[A] = new Decoder[A] {
+      override def decode(s: String): Int \/ A = f(s)
     }
 
-    // type aliases are needed to help with type inference, even kind-projector
-    // fails us...
-    type Out[a] = String \/ a
-    type MT[a] = ReaderT[Out, String, a]
     implicit val string: Decoder[String] = instance(_.right)
-    implicit val isoReaderT: Decoder <~> MT =
-      new IsoFunctorTemplate[Decoder, MT] {
-        def from[A](fa: MT[A]) = instance(fa.run(_))
-        def to[A](fa: Decoder[A]) = ReaderT[Out, String, A](fa.decode)
-      }
 
-    implicit val monad: MonadError[Decoder, String] = MonadError.fromIso(isoReaderT)
+    type I = String
+    type O[a] = Int \/ a
+    type MT[a] = ReaderT[O, I, a]
+    // type inference doesn't work
```

Comment:
I think Harrison is looking into why it's failing if we try to use the type directly instead of MT... it could be a compiler bug. If so, we'll put a link to the bug report here. It's super annoying that any type annotations are needed at all.

Reply:
Yeah I want to add a scala/bug ref to it so people know why there are so many type ascriptions. I'm trying to minimize the failure.

Reply:
Did you enable `-Ypartial-unification`?

Reply:
... do we actually not have it on anywhere? or am I just misreading sbt output?

I guess the idea is that we still support pre-2.12 scalas?

Especially for tests, is there a reason we don't use Miles' plugin?